stan/math/rev/functor/algebra_solver_newton.hpp

```diff
@@ -129,22 +129,29 @@ Eigen::Matrix<scalar_type_t<T2>, Eigen::Dynamic, 1> algebra_solver_newton(
 
   const auto& x_eval = x.eval();
   const auto& y_eval = y.eval();
+  const auto& x_val = (value_of(x_eval)).eval();
+  const auto& y_val = (value_of(y_eval)).eval();
   Eigen::VectorXd theta_dbl = algebra_solver_newton(
       f, x_eval, value_of(y_eval), dat, dat_int, msgs, scaling_step_size,
       function_tolerance, max_num_steps);
 
-  typedef system_functor<F, double, double, false> Fy;
-  typedef system_functor<F, double, double, true> Fs;
-  typedef hybrj_functor_solver<Fs, F, double, double> Fx;
-  Fx fx(Fs(), f, value_of(x_eval), value_of(y_eval), dat, dat_int, msgs);
+  typedef system_functor<F, double, double, true> Fsx;
+  typedef hybrj_functor_solver<Fsx, F, double, double> Fx;
+  Fx fx(Fsx(), f, value_of(x_eval), value_of(y_eval), dat, dat_int, msgs);
```

Comment:
Already have the value_ofs computed

Reply:
Ben how does this keep track of what vari's it needs to update?

Reply:
Not sure -- I didn't look that close.

Reply:
The way to figure this out I think would be to look at the stanc3 algebra_solver signature.

Reply:
I've precomputed the `value_of`s now. I won't resolve this yet, since I don't totally understand @SteveBronder's question. Are you worried about the vari inputs to the implicit function? (In any case, I think hopefully we'll resolve this tomorrow when we implement the new variadic signature for `algebra_solver`.)